### PR TITLE
fix: add server setup example

### LINKED ISSUE #191: config specific LS
i was using [nvim-lspinstall](https://github.com/kabouzeid/nvim-lspinstall). with it, i used to use following config for LUA
```lua

require'lspconfig'.lua.setup{ -- for Lua
  on_attach = on_attach,
  settings = {
    Lua = {
      diagnostics = {
        -- Get the language server to recognize the 'vim', 'use' global
        globals = {'vim', 'use'},
      },
      workspace = {
        -- Make the server aware of Neovim runtime files
        library = vim.api.nvim_get_runtime_file("", true),
      },
      -- Do not send telemetry data containing a randomized but unique identifier
      telemetry = {
        enable = false,
      },
    },
  }
}
```

how to do this with nvim-lsp-installer?

### PR DESCRIPTION
**Pull request by kylo252**
Add an example of how to configure the setup per server.

Fixes #191

### CODE REVIEW

**Comment by williamboman** [comment]
@kylo252 I pushed some changes to reflect the new `server:on_ready()` API, wdyt?

**Comment by kylo252** [comment]
> @kylo252 I pushed some changes to reflect the new `server:on_ready()` API, wdyt?

I saw that `opts` weren't obvious that they can be used with the new API, so I added them there. I also switched the variable-names for readability, but feel free to change it back if you think it looks odd.

File: README.md [hunk 1]
```diff
@@ -85,15 +87,19 @@ lsp_installer.on_server_ready(function(server)
...
-    if not rust_analyzer:is_installed() then
-        rust_analyzer:install()
+local server_available, requested_server = lsp_installer_servers.get_server("rust_analyzer")
+if server_available then
+    requested_server:on_ready(function ()
+        local opts = {}
+        requested_server:setup(opts)
+    end)
+    if not requested_server:is_installed() then
+        requested_server:install()
```
Comment (williamboman) [inline README.md:102]:
```suggestion
        -- Queue the server to be installed
        requested_server:install()
```

File: doc/nvim-lsp-installer.txt [hunk 2]
```diff
@@ -90,6 +86,24 @@ Then, somewhere in your initialization script (see `:h init.lua`): >
     end)
 <
 
+For more advanced use cases you may also interact with more APIs
+nvim-lsp-installer has to offer, for example the following (refer
+to `:help nvim-lsp-installer` for more docs): >
+
+    local lsp_installer_servers = require'nvim-lsp-installer.servers'
+
+    local server_avaiilable, requested_server = lsp_installer_servers.get_server("rust_analyzer")
```
Comment (williamboman) [inline doc/nvim-lsp-installer.txt:95]:
```suggestion
    local server_available, requested_server = lsp_installer_servers.get_server("rust_analyzer")
```

File: doc/nvim-lsp-installer.txt [hunk 3]
```diff
@@ -90,6 +86,24 @@ Then, somewhere in your initialization script (see `:h init.lua`): >
...
+    if server_avaiilable then
```
Comment (williamboman) [inline doc/nvim-lsp-installer.txt:96]:
```suggestion
    if server_available then
```

File: doc/nvim-lsp-installer.txt [hunk 4]
```diff
@@ -90,6 +86,24 @@ Then, somewhere in your initialization script (see `:h init.lua`): >
...
+        requested_server:on_ready(function ()
+            local opts = {}
+            requested_server:setup(opts)
+        end)
+        if not requested_server:is_installed() then
+            requested_server:install()
```
Comment (williamboman) [inline doc/nvim-lsp-installer.txt:102]:
```suggestion
             -- Queue the server to be installed
            requested_server:install()
```

**Review by williamboman — APPROVED** [review]

**Comment by williamboman** [comment]
Ty @kylo252 !!

**Pull request merged**